sql/copy: fix planning for vectorized COPY with nested renderNodes

cucaroach opened the pull request:
A COPY insert statement can produce a plan with zero or more renderNodes
and we were only allowing zero or one. An example of multiple render
nodes is a table with a hash shared primary key.

Fixes: #104662
Release note (sql change): Vectorized COPY FROM support didn't properly
handle schemas hash sharded primary keys.
Epic: none

cucaroach committed:
sql/copy: fix planning for vectorized COPY with nested renderNodes

A COPY insert statement can produce a plan with zero or more renderNodes
and we were only allowing zero or one. An example of multiple render
nodes is a table with a hash shared primary key.

Fixes: #104662
Release note (sql change): Vectorized COPY FROM support didn't properly
handle schemas hash sharded primary keys.
Epic: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111284)
<!-- Reviewable:end -->

cucaroach committed:
copy: add names to subtests for variations

Instead of names like TestDataDriven/copy_test#0 we can now use names
like:

TestDataDriven/copy_test/vectorize=off/atomic=on/fastPath=on/copy_from

Epic: none
Release note: None

yuzefovich reviewed (approved):
Nice find! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111284#-:-NfMdfFbAUC6LZxtA680:boyyrm)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @cucaroach)



<!-- Sent from Reviewable.io -->

cucaroach commented:
bors r+
